copy/copy.go

```diff
@@ -340,14 +340,26 @@ func (c *copier) copy(ctx context.Context, src, srcComponents, target string, ov
 
 	switch {
 	case fi.IsDir():
-		if created, err := c.copyDirectory(
+		created, err := c.copyDirectory(
 			ctx, src, srcComponents, target, fi, overwriteTargetMetadata,
 			include, includeMatchInfo, excludeMatchInfo,
-		); err != nil {
+		)
+		if err != nil {
 			return err
-		} else if !overwriteTargetMetadata || c.includePatternMatcher != nil {
+		}
+		if !overwriteTargetMetadata || c.includePatternMatcher != nil {
```

Comment:
The `|| c.includePatternMatcher != nil` here looks bogus. I think that should be removed.